mlrun/common/schemas/model_monitoring/functions.py

```diff
@@ -54,7 +54,11 @@ def from_function_dict(
 
         return cls(
             type=func_type,
-            name=func_dict["metadata"]["name"],
+            name=func_dict["metadata"]["name"]
+            if func_type != FunctionsType.APPLICATION
+            else func_dict["spec"]["graph"]["steps"]["PrepareMonitoringEvent"][
+                "class_args"
+            ]["application_name"],
```

Comment:
Maybe better to use get to prevent bug on the backend side
```suggestion
            else func_dict.get("spec",{}).get("graph",{}).get("steps", {}).get("PrepareMonitoringEvent", {}).get("class_args", {}).get("application_name", name)
```